web/packages/teleport/src/Roles/RoleEditor/StandardEditor.tsx

```diff
@@ -419,12 +437,166 @@ export function ServerAccessSpecSection({
+              })
+            }
+          />
+        ))}
+
+        <Box>
+          <ButtonSecondary
+            disabled={isProcessing}
+            gap={1}
+            onClick={() =>
+              onChange?.({
+                ...value,
+                resources: [...value.resources, newKubernetesResourceModel()],
+              })
+            }
+          >
+            <Icon.Add disabled={isProcessing} size="small" />
+            {value.resources.length > 0
+              ? 'Add Another Resource'
+              : 'Add a Resource'}
+          </ButtonSecondary>
+        </Box>
+      </Flex>
+    </>
+  );
+}
+
+function KubernetesResourceView({
+  value,
+  isProcessing,
+  onChange,
+  onRemove,
+}: {
+  value: KubernetesResourceModel;
+  isProcessing: boolean;
+  onChange(m: KubernetesResourceModel): void;
+  onRemove(): void;
+}) {
+  const { kind, name, namespace, verbs } = value;
+  const theme = useTheme();
+  return (
+    <Box
+      border={1}
+      borderColor={theme.colors.interactive.tonal.neutral[0]}
+      borderRadius={3}
+      padding={3}
+    >
+      <Flex>
+        <Box flex="1">
+          <H4 mb={3}>Resource</H4>
+        </Box>
+        <ButtonIcon
+          aria-label="Remove resource"
+          disabled={isProcessing}
+          onClick={onRemove}
+        >
+          <Icon.Trash
+            size="small"
+            color={theme.colors.interactive.solid.danger.default}
+          />
+        </ButtonIcon>
+      </Flex>
+      <FieldSelect
+        label="Kind"
+        isDisabled={isProcessing}
+        options={kubernetesResourceKindOptions}
+        value={kind}
+        onChange={k => onChange?.({ ...value, kind: k })}
+      />
+      <FieldInput
+        label="Name"
+        toolTipContent={
+          <>
+            Name of the resource. Special value <MarkInverse>*</MarkInverse>{' '}
+            means any name.
+          </>
+        }
+        disabled={isProcessing}
+        value={name}
+        onChange={e => onChange?.({ ...value, name: e.target.value })}
+      />
+      <FieldInput
+        label="Namespace"
+        toolTipContent={
+          <>
+            Namespace that contains the resource. Special value{' '}
+            <MarkInverse>*</MarkInverse> means any namespace.
+          </>
+        }
+        disabled={isProcessing}
+        value={namespace}
+        onChange={e => onChange?.({ ...value, namespace: e.target.value })}
+      />
+      <FieldSelect
+        isMulti
+        label="Verbs"
+        isDisabled={isProcessing}
+        options={kubernetesVerbOptions}
+        value={verbs}
+        onChange={v => onChange?.({ ...value, verbs: v })}
+        mb={0}
+      />
+    </Box>
+  );
 }
 
 export const EditorWrapper = styled(Box)<{ mute?: boolean }>`
   opacity: ${p => (p.mute ? 0.4 : 1)};
   pointer-events: ${p => (p.mute ? 'none' : '')};
 `;
+
+// TODO(bl-nero): This should ideally use tonal neutral 1 from the opposite
+// theme as background.
```

Comment:
Does the BBLP theme have an opposite theme? I wonder if each theme should have the concept of an inverse colors in itself.

Reply:
@ravicious We decided with Kenny that I'll later cherry-pick tonal neutral colors from each theme the inverse theme, treating light as inverse of BBLP. (Cherry-picking is the only option, since we can't just switch the color theme entirely for a subtree of components — exactly for the reason you described.) This should be a good solution until Kenny provides some more concrete guidelines on the `Mark` component itself.